### ISSUE TITLE
manifest deployer does not apply kind:StorageClass

### ISSUE THREAD

**Issue by fpwfhsi** [k0sproject/k0s#1427]
### Before creating an issue, make sure you've checked the following:

- [X] You are running the latest released version of k0s
- [X] Make sure you've searched for existing issues, both open and closed
- [X] Make sure you've searched for PRs too, a fix might've been merged already
- [X] You're looking at docs for the released version, "main" branch docs are usually ahead of released versions.

### Version

v1.22.5+k0s.0

### Platform

```shell
DISTRIB_ID="Flatcar Container Linux by Kinvolk"
DISTRIB_RELEASE=3033.2.0
DISTRIB_CODENAME="Oklo"
```

### What happened?

If I use manifest deployer to deploy a cloud csi driver, StorageClass from manifest is not created.
No matter what kind:StorageClass manifest is created under /var/lib/k0s/manifests/whatever_dir, is not created in kubernetes.

### Steps to reproduce

1. mkdir /var/lib/k0s/manifests/whatever_dir  && cat <<EOF > /var/lib/k0s/manifests/whatever_dir/storageclassnotcreated.yaml
> ---
kind: StorageClass
apiVersion: storage.k8s.io/v1
metadata:
  namespace: kube-system
  name: test-volumes
  annotations:
    storageclass.kubernetes.io/is-default-class: "true"
provisioner: csi.test.cloud
volumeBindingMode: WaitForFirstConsumer
allowVolumeExpansion: true
EOF

2. storage class does not appear after issuing command: k0s kubectl get storageclass
3. If I manually apply: k0s kubectl apply -f /var/lib/k0s/manifests/whatever_dir/storageclassnotcreated.yaml, it will appear.

### Expected behavior

Manifest deployer should create any kind of object which is present in manifest files.

### Actual behavior

Manifest deployer does not apply kind:StorageClass objects.

**Comment by fpwfhsi** [comment]
This bug is not present in v1.23.1+k0s.0, but in this version extension openebs is introduced, so maybe they are correlated.

**Comment by fpwfhsi** [comment]
I can't find anything relevant in logs.
I sent you an email with connection details maybe you can find something relevant.

regards

‐‐‐‐‐‐‐ Original Message ‐‐‐‐‐‐‐

**Comment by fpwfhsi** [comment]
I moved everything to helm

**Comment by till** [comment]
Would someone re-open this? I am having issues with the manifest controller (on Flatcar Linux) as well.

I am trying to use it to initialize my cloud-provider (openstack) and it won't apply anything I uploaded to my controller nodes. My worker nodes are not ready, until the cloud-provider is initialized.

I see that calico, metrics-server etc. manifests from `/var/lib/k0s/manifests` got applied (though they are in a restart loop as the worker nodes are not initialized). Is there anything I need to "manually" enable for the manifest deployer to work in this state? I am thinking it's a chicken/egg situation maybe.

**Comment by till** [comment]
Maybe related, when I apply the manifests through `sudo /opt/bin/k0s kubectl apply -f ...`:

```
sudo /opt/bin/k0s kubectl apply -f /var/lib/k0s/manifests/openstack/cloud-controller-manager-ds.yaml 
Warning: resource serviceaccounts/cloud-controller-manager is missing the kubectl.kubernetes.io/last-applied-configuration annotation which is required by kubectl apply. kubectl apply should only be used on resources created declaratively by either kubectl create --save-config or kubectl apply. The missing annotation will be patched automatically.
serviceaccount/cloud-controller-manager configured
Warning: resource daemonsets/openstack-cloud-controller-manager is missing the kubectl.kubernetes.io/last-applied-configuration annotation which is required by kubectl apply. kubectl apply should only be used on resources created declaratively by either kubectl create --save-config or kubectl apply. The missing annotation will be patched automatically.
daemonset.apps/openstack-cloud-controller-manager configured
```

**Comment by till** [comment]
I think I looked in the wrong place, but I can see that it actually applied my daemonset before but then it failed to start the pods because of a node selector in the definition. I think the section on "installing your own cloud-provider" is a bit too brief when it says, "consult the documentation" as for example the CP nodes not running workloads is very nice (and a huge plus) but also not very common.

So long story short, it seems like it's applying the manifests — but there is zero info in `journalctl -u k0scontroller` that it happened.
